smart-contracts/contracts/interfaces/IUnlock.sol

```diff
@@ -32,7 +32,7 @@ interface IUnlock {
     uint _maxNumberOfKeys
   )
     external
-    returns (ILockCore lock);
```

Comment:
We may need to change this again in the future...the idea here was that `createLock` would always return an `IlockCore`, whether it's a PublicLock, PrivateLock, etc... But. this can be addressed when we have more lock types I guess.